web3/_utils/module_testing/eth_module.py

```diff
@@ -758,6 +758,7 @@ def test_eth_getTransactionByBlockNumberAndIndex(
         assert is_dict(transaction)
         assert transaction['hash'] == HexBytes(mined_txn_hash)
 
+    @pytest.mark.xfail(raises=KeyError)
```

Comment:
This is the higest level that gets run for each different client (ethtester, geth, parity). Since only ethtester was failing, you'll need to remove this here. It will be overridden for ethtester where you have it above in `tests/integration/test_ethereum_tester.py`. Since we don't expect a failure for geth or parity (I don't think), you can remove the xfail here.